render/tests/test-event.js

```diff
@@ -884,4 +916,48 @@ o.spec("event", function() {
 			})
 		})
 	})
+	o("avoid sync redraw after removal", function() {
+		var div = m("div", {onclick: function () {}})
+		var e = $window.document.createEvent("MouseEvents")
+		e.initEvent("click", true, true)
+
+		render(root, div)
+		// remove div
+		render(root, [])
+
+		div.dom.dispatchEvent(e)
+
+		o(redraw.callCount).equals(0)
+	})
+	o("avoid async redraw after removal", function(done) {
+		var thenCB
+		var div = m("div", {onclick: async function () {
+			await new Promise(function(resolve){thenCB = resolve})
+		}})
+		var e = $window.document.createEvent("MouseEvents")
+		e.initEvent("click", true, true)
+
+		render(root, div)
+		div.dom.dispatchEvent(e)
+
+		o(redraw.callCount).equals(1)
+		o(redraw.this).equals(undefined)
+		o(redraw.args.length).equals(0)
```

Comment:
Can you also assert `thenCB` is set? That being set also implies that the event listener was actually called.

Reply:
OK, added the following assertions.
- Whether the event listener itself has been called or not.
- Whether thenCB was set or not.